src/Microsoft.AspNetCore.Razor/Editor/ImplicitExpressionEditHandler.cs

```diff
@@ -149,10 +154,65 @@ private static bool IsSecondaryDotlessCommitInsertion(Span target, TextChange ch
                    change.OldLength == 0;
         }
 
-        private static bool IsAcceptableReplace(Span target, TextChange change)
+        private bool IsAcceptableReplace(Span target, TextChange change)
         {
             return IsEndReplace(target, change) ||
-                   (change.IsReplace && RemainingIsWhitespace(target, change));
+                (change.IsReplace && RemainingIsWhitespace(target, change));
+        }
+
+        private bool IsAcceptableIdentifierReplacement(Span target, TextChange change)
+        {
+            if (!change.IsReplace)
+            {
+                return false;
+            }
+
+            for (var i = 0; i < target.Symbols.Count; i++)
```

Comment:
Yeah, a binary search should work since it should be ordered. But I agree we should see how this is to start with and get the tests right. I think a binary search is a pretty simple optimization to add in once we've got the code correct though, and it'll pay off quite a bit.